### PR TITLE
Restricts cell identification ellipses to annotated regions

### PR DESCRIPTION
**Pull request by JABull1066**
Only draws ellipses if ellipse centroid is inside annotated region. (Hopefully)

### CODE REVIEW

File: src/util/TileCellData.js [hunk 1]
```diff
@@ -182,11 +189,29 @@ export default class TileCellData {
       if (cell_area > 10) {
         // fit an ellipse
         const rotatedRect = cv.fitEllipse(contours.get(i));
-        // add ellipse to data
-        this.add_cell(rotatedRect.center.x, rotatedRect.center.y, rotatedRect.size.width, rotatedRect.size.height, rotatedRect.angle);
+
+        // Need to map point (in tile coords) to annotation (in % of image)
+        let scale = this.tile.bounds.height / this.tile.sourceBounds.height;
+
+        // let shift_x = (this.tile.x) * this.tile.bounds.width;
```
Comment (martinjrobins) [inline src/util/TileCellData.js]: don't think this is needed anymore?

File: src/util/TileCellData.js [hunk 2]
```diff
@@ -182,11 +189,29 @@ export default class TileCellData {
...
+        // let shift_y = (this.tile.y) * this.tile.bounds.height;
+        let shift_x = this.tile.x * this.tile.bounds.x / this.tile.x;
+        let shift_y = this.tile.y * this.tile.bounds.y / this.tile.y;
+
+        // test if ellipse centroid is contained in the annotation polygon
```
Comment (martinjrobins) [inline src/util/TileCellData.js]: this comment doesn't refer to the line below, probably should be just above the `if` statement I think

**Review by martinjrobins — CHANGES_REQUESTED** [review]
looks great, thanks @JABull1066, just need to tidy up the comments as per suggestions below

**Comment by martinjrobins** [comment]
hi @JABull1066, how are you going on this? Is it almost ready to merge?

**Comment by JABull1066** [comment]
Almost - I didn't quite manage to get everything done before getting pulled on to paper writing, unfortunately! I just need to find a free afternoon...could we add an extra day in to the week?

**Review by martinjrobins — APPROVED** [review]
great, thanks @JABull1066

**Pull request merged**